samza-sql/src/main/java/org/apache/samza/sql/interfaces/SourceResolver.java

```diff
@@ -19,26 +19,21 @@
 
 package org.apache.samza.sql.interfaces;
 
+
 /**
- * Source Resolvers are used by Samza Sql application to fetch the {@link SqlSystemSourceConfig} corresponding to the source.
+ * Source Resolvers are used by Samza Sql application to fetch the {@link SqlSystemSourceConfig} corresponding
+ * to the source. Additionally, it provides capability to resolve Samza table sources and creating the matching
+ * table descriptors.
  */
 public interface SourceResolver {
   /**
    * Returns the SystemStream config corresponding to the source name
    * @param sourceName
    *  source whose systemstreamconfig needs to be fetched.
+   * @param isSink
+   *  whether the source is an input or output source in the SQL statement
    * @return
    *  System stream config corresponding to the source.
    */
-  SqlSystemSourceConfig fetchSourceInfo(String sourceName);
-
-  /**
-   * Returns if a given source is a table. Different source resolvers could have different notations in the source
-   * name for denoting a table. Eg: system.stream.$table
-   * @param sourceName
-   *  source that needs to be checked if it is a table.
-   * @return
-   *  true if the source is a table, else false.
-   */
-  boolean isTable(String sourceName);
+  SqlSystemSourceConfig fetchSourceInfo(String sourceName, boolean isSink);
```

Comment:
Do you think it is better to have two methods here fetchSourceInfo and fetchSinkInfo?

Reply:
I think that makes sense.